### PR TITLE
notifier filter for completed trades

### PR DESCRIPTION
**Pull request by MCYBA**
when ı use notifer for seing my trades data, canceled orders become so big waste, then ı create a filter to zenbot for sending notifier only completed messages.

### CODE REVIEW

File: commands/trade.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ module.exports = function (program, conf) {
     .option('--use_fee_asset', 'Using separated asset to pay for fees. Such as binance\'s BNB or Huobi\'s HT', Boolean, false)
     .option('--run_for <minutes>', 'Execute for a period of minutes then exit with status 0', String, null)
     .option('--debug', 'output detailed debug info')
+    .option('--only_completed_trades', 'Message filter for only completed trades to notifier.', Boolean, conf.notifiers.only_completed_trades)
```
Comment (DeviaVir) [inline commands/trade.js]: I don't think we should add this, but just keep it as a `conf` option instead.

File: lib/engine.js [hunk 2]
```diff
@@ -425,8 +425,12 @@ module.exports = function (s, conf) {
               }, conf.wait_for_settlement)
             }
             else {
-              pushMessage('Buying ' + formatAsset(size, s.asset) + ' on ' + s.exchange.name.toUpperCase(), 'placing buy order at ' + formatCurrency(price, s.currency) + ', ' + formatCurrency(quote.bid - Number(price), s.currency) + ' under best bid\n')
-              doOrder()
+              if(!so.only_completed_trades){
```
Comment (DeviaVir) [inline lib/engine.js]: this should probably be 
```suggestion
              if(!conf.notifiers.only_completed_trades){
```

File: lib/engine.js [hunk 3]
```diff
@@ -473,8 +477,12 @@ module.exports = function (s, conf) {
               }, conf.wait_for_settlement)
             }
             else {
-              pushMessage('Selling ' + formatAsset(size, s.asset) + ' on ' + s.exchange.name.toUpperCase(), 'placing sell order at ' + formatCurrency(price, s.currency) + ', ' + formatCurrency(Number(price) - quote.bid, s.currency) + ' over best ask\n')
-              doOrder()
+              if(!so.only_completed_trades){
```
Comment (DeviaVir) [inline lib/engine.js]:
```suggestion
              if(!conf.notifiers.only_completed_trades){
```

File: commands/trade.js [hunk 4]
```diff
@@ -76,6 +76,7 @@ module.exports = function (program, conf) {
           so[k] = cmd[k]
         }
       })
+      so.only_completed_trades = conf.only_completed_trades
```
Comment (DeviaVir) [inline commands/trade.js]: I think you can leave this out entirely.
```suggestion
      
```

**Review by DeviaVir — CHANGES_REQUESTED** [review]
some more things

**Comment by MCYBA** [comment]
Mister, sorry about i taken your time. i don't understand, there is no problem on my repository. Could you explain me about this case?

**Review by DeviaVir — APPROVED** [review]

**Pull request merged**